src/provider/mod.rs

```diff
@@ -347,12 +333,48 @@ macro_rules! impl_traits {
         self.as_ref().to_vec()
       }
     }
+  };
+}
 
-    impl CompressedGroup for $name_compressed {
-      type GroupElement = $name::Point;
+/// Nova folding circuit engine and curve group ops
+#[macro_export]
+macro_rules! impl_folding {
```

Comment:
Is there a reason to call this `impl_folding`? It seems more appropriate to call it `impl_engine`?

Reply:
That makes more sense to me.
I renamed accordingly.